### PR TITLE
GH-41527: [CI][Dev] Remove unncessary requirements for six

### PR DESCRIPTION
**Pull request by llama90**
### Rationale for this change

The `six` requirement is a compatibility library for Python 2 and 3. Since Arrow no longer supports Python 2, we need to remove `six` requirement from merge script ([Python compatibility](https://arrow.apache.org/docs/python/install.html#python-compatibility)).

### What changes are included in this PR?

- Removing `six` requirement
- Updating `merge_arrow_pr.py`
  - `six.string_types` to `str`
  - `six_binary_type` to `bytes`

### Are these changes tested?

Tested by CI.

### Are there any user-facing changes?

Maybe not. It is for developers.

* GitHub Issue: #41527

### CODE REVIEW

File: dev/requirements_merge_arrow_pr.txt [hunk 1]
```diff
@@ -1,3 +1,2 @@
 jira
```
Comment (llama90) [inline dev/requirements_merge_arrow_pr.txt:1]: Is JIRA still supported?

Comment (assignUser) [reply dev/requirements_merge_arrow_pr.txt:1]: It is no longer supported as the issues where moved from jira to arrow gh issues: https://lists.apache.org/thread/01f570w32df5prr8tqwrsq5nq4vn570x

But we should probably wait with the removal until after 17.0.0 and remove all jira stuff from the release scritps as well or what do you think @raulcd

**Review by kou — APPROVED** [review]
+1

File: dev/requirements_merge_arrow_pr.txt (hunk 1 above)
Comment (raulcd) [reply dev/requirements_merge_arrow_pr.txt:1]: I agree, better wait for 17.0.0 and remove all the JIRA related stuff both on the merge script, archery, etc

**Pull request merged**